ui/components-react/src/test/filter-builder/FilterBuilderState.test.ts

```diff
@@ -684,14 +692,160 @@ describe("usePropertyFilterBuilder", () => {
+          ).to.be.eq(
+            UiComponents.translate("filterBuilder.errorMessages.emptyValue")
+          );
+        });
+
+        it("returns undefined and sets rule error message to `Value is empty` if item`s range `to` value is empty", () => {
+          const { result } = renderHook(() =>
+            usePropertyFilterBuilder({
+              initialFilter: createRangeFilter(
+                property,
+                "between",
+                {
+                  valueFormat: PropertyValueFormat.Primitive,
+                  value: 1,
+                },
+                {
+                  valueFormat: PropertyValueFormat.Primitive,
+                }
+              ),
+            })
+          );
+          const { buildFilter } = result.current;
+
+          const buildFilterResult = buildFilter();
+          expect(buildFilterResult).to.be.undefined;
+
+          const { rootGroup } = result.current;
+          expect(
+            (rootGroup.items[0] as PropertyFilterBuilderRule).errorMessage
+          ).to.be.eq(
+            UiComponents.translate("filterBuilder.errorMessages.emptyValue")
+          );
+        });
+
+        it("returns undefined and sets rule error message to `Invalid range` if item`s range is not valid", () => {
+          const { result } = renderHook(() =>
+            usePropertyFilterBuilder({
+              initialFilter: createRangeFilter(
+                property,
+                "between",
+                {
+                  valueFormat: PropertyValueFormat.Primitive,
+                  value: 2,
+                },
+                {
+                  valueFormat: PropertyValueFormat.Primitive,
+                  value: 1,
+                }
+              ),
+            })
+          );
+          const { buildFilter } = result.current;
+
+          const buildFilterResult = buildFilter();
+          expect(buildFilterResult).to.be.undefined;
+
+          const { rootGroup } = result.current;
+          expect(
+            (rootGroup.items[0] as PropertyFilterBuilderRule).errorMessage
+          ).to.be.eq(
+            UiComponents.translate("filterBuilder.errorMessages.invalidRange")
+          );
+        });
+
+        it("returns property filter with `Between` rule when value is valid", () => {
+          const initialFilter = createRangeFilter(
+            property,
+            "between",
+            {
+              valueFormat: PropertyValueFormat.Primitive,
+              value: 1,
+            },
+            {
+              valueFormat: PropertyValueFormat.Primitive,
+              value: 2,
+            }
+          );
+          const { result } = renderHook(() =>
+            usePropertyFilterBuilder({
+              initialFilter,
+            })
+          );
+          const { buildFilter } = result.current;
+
+          const buildFilterResult = buildFilter();
+          expect(buildFilterResult).to.not.be.eq(initialFilter);
```

Comment:
why are we expecting them to **not** match? maybe the check should be more specific about what the result should be?